derive/src/taptree.rs

```diff
@@ -112,6 +120,13 @@ impl<'a> IntoIterator for &'a TapTree {
 }
 
 impl TapTree {
+    pub fn with_single_leaf(leaf: impl Into<LeafScript>) -> TapTree {
+        Self(vec![LeafInfo {
+            depth: u7::ZERO,
```

Comment:
This is really a u7?

Reply:
Yes, since taptree can have up to 127 levels (and up to 2^127 leafs)